gematria/datasets/find_accessed_addrs_exegesis.cc

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gematria/datasets/find_accessed_addrs_exegesis.h"
+
+#include "BenchmarkRunner.h"
+#include "LlvmState.h"
+#include "Target.h"
+#include "TargetSelect.h"
+#include "X86.h"
+#include "X86InstrInfo.h"
+#include "X86RegisterInfo.h"
+#include "gematria/llvm/disassembler.h"
+
+using namespace llvm;
+using namespace llvm::exegesis;
+
+namespace gematria {
+
+ExegesisAnnotator::ExegesisAnnotator(
+    LlvmArchitectureSupport &ArchSupport_, LLVMState &State_,
+    std::unique_ptr<BenchmarkRunner> Runner_,
+    std::unique_ptr<const SnippetRepetitor> Repetitor_)
+    : ArchSupport(ArchSupport_),
+      MCPrinter(ArchSupport.CreateMCInstPrinter(0)),
+      State(State_),
+      Runner(std::move(Runner_)),
+      Repetitor(std::move(Repetitor_)) {}
+
+Expected<std::unique_ptr<ExegesisAnnotator>> ExegesisAnnotator::Create(
+    LlvmArchitectureSupport &ArchSupport_, LLVMState &State_) {
+  // Initialize the supported Exegesis targets. Currently we only support X86.
+  InitializeX86ExegesisTarget();
+
+  if (pfm::pfmInitialize())
+    return make_error<StringError>(
+        "Failed to initialize libpfm",
+        std::make_error_code(std::errc::invalid_argument));
+
+  auto RunnerOrErr = State_.getExegesisTarget().createBenchmarkRunner(
+      Benchmark::Latency, State_, BenchmarkPhaseSelectorE::Measure,
+      BenchmarkRunner::ExecutionModeE::SubProcess, 1, Benchmark::Min);
+
+  if (!RunnerOrErr) return RunnerOrErr.takeError();
+
+  std::unique_ptr<const SnippetRepetitor> Repetitor_ =
```

Comment:
No trailing underscore in the variable name.

Reply:
Fixed